SU2_CFD/include/iteration/CIteration.hpp

```diff
@@ -132,12 +132,23 @@ class CIteration {
    * \param[in] FFDBox - FFD FFDBoxes of the problem.
    * \param[in] val_iZone - Index of the zone.
    * \param[in] val_iInst - Index of the instantiation.
-   * \param[in] CrossTerm - Boolean for CrossTerm.
    */
   virtual void Iterate(COutput* output, CIntegration**** integration, CGeometry**** geometry, CSolver***** solver,
                        CNumerics****** numerics, CConfig** config, CSurfaceMovement** surface_movement,
                        CVolumetricMovement*** grid_movement, CFreeFormDefBox*** FFDBox, unsigned short val_iZone,
-                       unsigned short val_iInst, bool CrossTerm){}
+                       unsigned short val_iInst){}
+
+  /*!
+   * \brief A virtual member.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver - Container vector with all the solutions.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] val_iZone - Index of the zone.
+   * \param[in] val_iInst - Index of the instantiation.
+   * \param[in] CrossTerm - Boolean for CrossTerm.
+   */
+  virtual void IterateDiscAdj(CGeometry**** geometry, CSolver***** solver, CConfig** config,
+                              unsigned short val_iZone, unsigned short val_iInst, bool CrossTerm) {}
```

Comment:
While renaming we could introduce what it actually does: e.g. `ExtractAdjontsOfInputs(...)` (+residual computation and relaxation). It calls `ExtractAdjoint_Solution/Variable` which is pretty much the same name but that's it. 
As it is just fluid, heat, and FEA (for now) one could also think of ditching the `<physics>Iteration.cpp` implementation and do it once here maybe in CIteration(.cpp) now that it is its own func? Just thinking loud here, there are prob reasons against :)